src/pages/public/movies/MovieDetailPage.vue

```diff
@@ -21,59 +20,46 @@ export default defineComponent({
       required: true,
     },
   },
-  setup() {
-    // MovieDetailPage title should be dynamic, but we have some errors here
-    // please check the comments below
-    useMeta({ title: 'Movie Details' });
-    //
-  },
-  data() {
-    return {
-      isLoading: true,
-      storedMovie: null,
-    };
-  },
-  async mounted() {
-    if (this.allMovies.length > 0) {
-      this.findMovieInAllMovies();
-    } else {
-      await this.loadSingleMovie();
-    }
-  },
-  computed: {
-    movieTitle() {
-      return this.isLoading ? '' : this.storedMovie.title;
-    },
-    ...mapState(mainStore, ['allMovies', 'areMoviesLoading']),
-  },
-  methods: {
-    findMovieInAllMovies() {
-      const filteredMovie = this.allMovies.find(movie => movie.id == this.movieId);
-
-      this.storedMovie = filteredMovie;
-      this.isLoading = false;
+  setup(props) {
+    const { allMovies, showError } = useMainStore();
+    const isLoading = ref(true);
+    const storedMovie = ref(null);
+    const movieTitle = computed(() => (isLoading.value ? '' : storedMovie.value.title));
+    const setTitle = title => useMeta({ title: title });
 
-      // This works:
-      //
-      // useMeta({ title: this.storedMovie.title });
-      //
-    },
+    const findMovieInAllMovies = () => {
+      const filteredMovie = allMovies.find(movie => movie.id == props.movieId);
+      storedMovie.value = filteredMovie;
+      isLoading.value = false;
+    };
 
-    async loadSingleMovie() {
-      this.isLoading = true;
+    const loadSingleMovie = async () => {
+      isLoading.value = true;
       try {
-        this.storedMovie = await getMovieData(this.movieId);
-      } catch {
-        this.$router.push({ name: '404Page' });
+        storedMovie.value = await getMovieData(props.movieId);
+      } catch (error) {
+        showError(error);
       } finally {
-        this.isLoading = false;
+        isLoading.value = false;
+      }
+    };
 
-        // But this line throws error: No manager or current instance at useMeta
-        //
-        // useMeta({ title: this.storedMovie.title });
-        //
+    onMounted(async () => {
+      if (allMovies.length > 0) {
+        findMovieInAllMovies();
+        setTitle(movieTitle.value); // this works well
+      } else {
+        await loadSingleMovie();
       }
-    },
+      // --- TODO: This is still giving problems in / after "else" scenario ---
+      // setTitle(movieTitle.value); // >>> "Error: No manager or current instance"
```

Comment:
The reason for that is you shouldn't call `useMeta` multiple times but only once - and pass a computed there which would always return all values up to date for the meta fields 😄

Reply:
Thank you!

Reply:
There is still some issue with using `useMeta()` directly after loading data on mounted, even though the data-loading is asynchronous and awaits for API response data, and also `useMeta()` is called only once for sure. But! that worked:
```
const movieTitle = computed(() => (isLoading.value ? '' : storedMovie.value.title));

// ...

onMounted(async () => {
  if (allMovies.length > 0) {
    findMovieInAllMovies();
  } else {
    await loadSingleMovie();
  }
});

onUpdated(() => {
  useMeta({ title: movieTitle.value });
});
```
As updated hook will run always after we are sure our computed `movieTitle` has changed from an empty string. I don't know what exactly the issue is but for me it looks like `useMeta()` does not like such async usage.

What do you think?

Reply:
Have you tried using it as follows ?

```js
const meta = computed(() => {
  return {
    title: isLoading.value ? '' : storedMovie.value.title 
  }
});

useMeta(meta);
 ```

Reply:
Now that I look at it - you are using quite a dated package for this, I might recommend using `@vueuse/head` instead:
https://github.com/vueuse/head

Reply:
If the above doesn't work that is!

Reply:
Now I did and it works perfectly 😅 
Thank you!
So the thing is to return a whole object in a computed, so the object is reactive, and not its single property as I tried to do?

Reply:
And as far as I remember I used `vue-meta` because someone recommended it in one of our meetings.

Reply:
It could be a part of SEO & SRR presentation

Reply:
Yeah my bad, now that I see it just... wasn't updated for a while! 😮 And documentation seems to be lost for such usecase - passing in a computed 😅 

Thanks to the computed and it being reactive - the meta will now update properly ❤️

Reply:
Alright, so I think it's ready to be re-reviewed 😄

Reply:
Done and approved 💚